iib/workers/tasks/build.py

```diff
@@ -310,8 +311,13 @@ def _get_resolved_image(pull_spec):
     :rtype: str
     """
     log.debug('Resolving %s', pull_spec)
-    skopeo_output = skopeo_inspect(f'docker://{pull_spec}')
-    pull_spec_resolved = f'{skopeo_output["Name"]}@{skopeo_output["Digest"]}'
+    if '@' in pull_spec:
+        name = pull_spec.rsplit('@', 1)[0]
+    else:
+        name = pull_spec.rsplit(':', 1)[0]
+    skopeo_output = skopeo_inspect(f'docker://{pull_spec}', '--raw', return_json=False)
+    digest = hashlib.sha256(skopeo_output.encode('utf-8')).hexdigest()
```

Comment:
This is probably ok.

A little background. Ideally, the digest can be pulled from the API request header. However, this would mean teaching IIB how to handle registry auth directly, and that will not work with registries that don't set the header, i.e. Pulp.